src/unix/filesystem.c

```diff
@@ -0,0 +1,150 @@
+#include <unix_internal.h>
+#include <filesystem.h>
+
+#define SYMLINK_HOPS_MAX    8
+
+// fused buffer wrap, split, and resolve
+int resolve_cstring(tuple cwd, const char *f, tuple *entry, tuple *parent)
+{
+    if (!f)
+        return -EFAULT;
+
+    tuple t = *f == '/' ? filesystem_getroot(current->p->fs) : cwd;
+    tuple p = t;
+    buffer a = little_stack_buffer(NAME_MAX);
+    char y;
+    int nbytes;
+    int err;
+
+    while ((y = *f)) {
+        if (y == '/') {
+            if (buffer_length(a)) {
+                p = t;
+                t = lookup(t, intern(a));
+                if (!t)
+                {
```

Comment:
inconsistent formatting

Reply:
I used this formatting to minimize the diff lines of the commit that adds these parentheses, but OK, will fix it.